packages/@react-stately/collections/src/TreeCollection.ts

```diff
@@ -18,9 +18,12 @@ export class TreeCollection<T> implements Collection<Node<T>> {
   private iterable: Iterable<Node<T>>;
   private firstKey: Key;
   private lastKey: Key;
+  private disabledKeys: Set<Key>
 
-  constructor(nodes: Iterable<Node<T>>, expandedKeys: Set<Key>) {
+  constructor(nodes: Iterable<Node<T>>, {expandedKeys, disabledKeys}: {expandedKeys?: Set<Key>, disabledKeys?: Set<Key>} = {}) {
     this.iterable = nodes;
+    this.disabledKeys = disabledKeys || new Set();
+    expandedKeys = expandedKeys || new Set();
```

Comment:
I'm not 100% sure, but I noticed that the disabledKeys for other components like ListBox, Grid, etc are handled within state and that their respective keyboard delegates handle the logic for skipping those disabledKeys (https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-aria/grid/src/GridKeyboardDelegate.ts#L102, https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-aria/selection/src/ListKeyboardDelegate.ts#L34, and https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-stately/collections/src/ListLayout.ts#L342)

Perhaps the fix for ActionGroup would be to replicate the same kind of logic in the ActionGroupDelegate: https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-aria/actiongroup/src/ActionGroupKeyboardDelegate.ts

Reply:
Then you could pass `state.disabledKeys` to the delegate here: https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-aria/actiongroup/src/useActionGroup.ts#L51

Reply:
I agree with Daniel. I think the disabledKeys logic needs to be handled in the keyboard delegate. Sometimes we use the collection to traverse all of the items, and this would make it appear like the disabled items did not exist at all.